lambda-layer/build-layer.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+SOURCEDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+
+## revert https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7970
+OTEL_VERSION="1.32.1"
+ADOT_VERSION="1.32.1"
+
+
+## Clone and Patch the OpenTelemetry Java Instrumentation Repository
+git clone https://github.com/open-telemetry/opentelemetry-java-instrumentation.git
+pushd opentelemetry-java-instrumentation
+git checkout v${OTEL_VERSION} -b tag-v${OTEL_VERSION}
+
+# This patch is for Lambda related context propagation
+patch -p1 < "$SOURCEDIR"/patches/opentelemetry-java-instrumentation.patch
+
+# There is another patch in the .github/patches directory for other changes. We should apply them too for consistency.
+patch -p1 < "$SOURCEDIR"/../.github/patches/opentelemetry-java-instrumentation.patch
+
+git add -A
+git commit -m "Create patch version"
+./gradlew publishToMavenLocal
```

Comment:
After this step, the versions in zip file are ?
```
OTEL_VERSION="1.32.1"
ADOT_VERSION="1.32.1"
```

Reply:
Yes. Verified from the following line in the CW logs for Lambda function execution. The version of the javaagent .jar is `1.32.1-adot-lambda1` which is specified while building the layer and `-aws` is suffixed afterwards by the agent itself.
```
[otel.javaagent 2024-11-19 17:50:53:476 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.32.1-adot-lambda1-aws
```

Reply:
Is it possible to refer to version in https://github.com/aws-observability/aws-otel-java-instrumentation/blob/b58b94f1dd64d8f4b7d5658afc009d2bea0567a3/.github/patches/versions

Reply:
It is. Updated the `build-layer.sh` script to pull the otel version from the above mentioned file.
It is worth noting that even though the script can get the version from this file, the patches to be applied need the version to be hard coded in them.